src/main/java/edu/umich/its/cpm/MigrationController.java

```diff
@@ -246,4 +253,63 @@ public Migration migrated(HttpServletRequest request) {
 		}
 		return rv;
 	}
+	
+	/************* Box integration *****************/
+	/**
+	 * get json string og box folders
+	 * @return
+	 */
+	private String DEVELOPER_TOKEN = "";
+	private static final int MAX_DEPTH = 1;
+    
+	@RequestMapping("/box/folders")
+	public String getBoxFolders() {
+		
+		String rv = "";
+
+		// TODO: get the DEVELOPER_TOKEN from configuratio for now, will do token auto-generation in the future
+		DEVELOPER_TOKEN = env.getProperty("box_api_token");
+		
+		BoxAPIConnection api = new BoxAPIConnection(DEVELOPER_TOKEN);
+		BoxUser.Info userInfo = BoxUser.getCurrentUser(api).getInfo();
+		log.info("Box user log in success: user name = " + userInfo.getName() + " user login = " + userInfo.getLogin());
+		
+		BoxFolder rootFolder = BoxFolder.getRootFolder(api);
+		List<BoxFolder> folders = listFolder(rootFolder, 0);
+		
+		// create a new Gson instance
+		 Gson gson = new Gson();
+		 // convert BoxFolder list to json
+		 rv = gson.toJson(folders);
```

Comment:
The URL is rather a path in the JSON feed. I have attached the sample JSON inside JIRA